### DISCUSSION TITLE
Cant Install Postal

### DISCUSSION THREAD

**Discussion by easyemail** [postalserver/postal#2051] [Help with using Postal]
Hello! Hoping someone here may be able to help me. I have tried installing postal 3 different time now and still am having trouble. I am not very familiar with ubuntu, so I am likely missing something or messing something up. I have followed 3 different tutorials on this entire process. 1 YouTube video, and 2 stand alone installation topics-

https://computingforgeeks.com/install-and-configure-postal-mail-server-on-ubuntu/
and 
https://docs.postalserver.io/install/installation

The first spot I am seeing an error is when I try to make the initial Postal Admin User. I get an error-

postal make-user
Postal User Creator
Enter the information required to create a new Postal user.
This tool is usually only used to create your initial admin user.

/usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `connect': Access denied for user 'root'@'<redacted>' (using password: YES) (Mysql2::Error::ConnectionError)
        from /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:90:in `initialize'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `new'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/mysql2_adapter.rb:22:in `mysql2_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:830:in `new_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:874:in `checkout_new_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:853:in `try_to_checkout_new_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:814:in `acquire_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:538:in `checkout'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:382:in `connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:1033:in `retrieve_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_handling.rb:118:in `retrieve_connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/connection_handling.rb:90:in `connection'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/model_schema.rb:466:in `load_schema!'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/attributes.rb:234:in `load_schema!'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/attribute_decorators.rb:51:in `load_schema!'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/model_schema.rb:459:in `block in load_schema'
        from /usr/local/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/model_schema.rb:456:in `load_schema'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/model_schema.rb:346:in `attribute_types'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/attribute_methods.rb:230:in `has_attribute?'
        from /usr/local/bundle/gems/activerecord-5.2.6.2/lib/active_record/inheritance.rb:55:in `new'
        from /opt/postal/app/lib/postal/user_creator.rb:12:in `start'
        from script/make_user.rb:7:in `<main>'
        
I am guessing if I dont make the user, I likely cant access my postal web dashboard right? I am guessing this because when I tried to install postal the first time, I tried going to my domain and it was saying that I couldnt access the site.

When i did the second install, I did postal.mydomain.com and I got to a screen that said "We're sorry, but something went wrong.". I saw some other users on here having that issue, and they restarted mysql, ngnix and postal services. I have no idea what ngnix is, and wen I tried to restart it, it said it didnt exist.

I think there is something wrong with my mysql user permissions, but I cant seem to grant those permissions. Any help is massively appreciated as I am ready to pull my hair out.

**Comment by willpower232** [comment]
> Access denied for user 'root'@'<redacted>' (using password: YES)

This means the details you've written in the config file are incorrect. You should have a look at making a Postal-specific user for your databases rather than using your root details.
